src/main/java/nextstep/subway/line/domain/LineStation/LineStations.java

```diff
@@ -97,8 +98,22 @@ private boolean firstStation(Station station) {
 
     public void delete(Long stationId) {
         validationDelete();
-        Optional<LineStation> lineStation = findDeleteByLineStation(stationId);
+        Optional<LineStation> preStation = findDeleteByPreStation(stationId);
 
+        if (!preStation.isPresent()) {
+            Optional<LineStation> nextStation = findDeleteByNextStation(stationId);
+            deleteByNextStation(nextStation);
+        }
+
+        deleteByPreStation(preStation);
+    }
```

Comment:
`if (!preStation.isPresent()) `
하행역이 일치하는 경우에도 `deleteByPreStation` 까지 수행될 거 같아요 
이 때문에 deleteByPreStation 의 경우, 삭제 대상 역이 존재 하지 않는 경우 예외를 발생시키지 않고 
null을 remove 하는 것으로 보이는데요, 맞을까요..? 

아래와 같이 수정하고, deleteByPreStation 에서도 삭제 대상 역이 없다면 예외를 발생시키는게 좋을 것 같아요1

```suggestion
        if (!preStation.isPresent()) {
            Optional<LineStation> nextStation = findDeleteByNextStation(stationId);
            return deleteByNextStation(nextStation);
            
            // 또는
            deleteByNextStation(nextStation);
            return;
        }

        deleteByPreStation(preStation);
    }
```